app/Utils/UserFilter.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace App\Utils;
+
+use App\Models\Semester;
+use App\Models\SemesterStatus;
+use App\Models\User;
+use Illuminate\Database\Eloquent\Builder;
+
+/**
+ * A class that can be used to filter {@see User} instances in queries based on various criteria.
+ *
+ * Usage:
+ * <ol>
+ * <li>Create a user query that should be further narrowed down, e.g. {@see User::canView()}</li>
+ * <li>Create a new class instance with the previously obtained query instance</li>
+ * <li>Call the desired filter methods on the instance to narrow down the query</li>
+ * <li>Get the final query by calling {@see UserFilter::getQuery()} on the instance</li>
+ * </ol>
+ */
+class UserFilter
```

Comment:
The functionality of this class can be achieved by Laravel's scopes (see https://laravel.com/docs/10.x/eloquent#local-scopes). Rewrite it with scopes and try to create as general scopes as possible.

Reply:
Thanks!